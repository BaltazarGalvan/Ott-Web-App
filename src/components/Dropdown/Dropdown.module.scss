@use '../../styles/variables';
@use '../../styles/theme';
@use '../../styles/mixins/responsive';

$select-border: variables.$gray-lighter;
$select-arrow: variables.$white;
$select-focus: theme.$primary-color;

.dropdown {
  position: relative;
  display: grid;
  align-items: center;
  min-width: 10ch;
  max-width: 20ch;
  margin-left: 24px;
  padding: 0.2em 0.4em;
  font-size: 1.25rem;
  line-height: 1.1;
  background-color: rgba(0, 0, 0, 0.7);
  border: 1px solid rgba(255, 255, 255, 0.3);
  border-radius: 0.25em;
  cursor: pointer;
  grid-template-areas: 'select';

  // Custom arrow
  &::after {
    z-index: 1;
    width: 0.4em;
    height: 0.3em;
    background-color: $select-arrow;
    clip-path: polygon(100% 0%, 0 0%, 50% 100%);
    content: '';
    justify-self: end;
  }

  .select {
    z-index: 1;
    width: 100%;
    margin: 0;
    padding: 0 1em 0 0;
    color: var(--highlight-color, white);
    font-family: inherit;
    font-weight: 700;
    font-size: 1rem;
    line-height: inherit;
    background-color: rgba(0, 0, 0, 1);
    border: none;
    outline: none;
    cursor: inherit;
    appearance: none;

    // Remove IE arrow
    &::-ms-expand {
      display: none;
    }
  }

  .select,
  &::after {
    grid-area: select;
  }

  & .disabled {
    background-color: #eee;
    background-image: linear-gradient(to top, #ddd, #eee 33%);
    cursor: not-allowed;
  }
  &:hover {
    border-color: $select-focus;
  }

  &:focus + .focus {
    position: absolute;
    top: -1px;
    right: -1px;
    bottom: -1px;
    left: -1px;
    border: 2px solid orange;
    border-radius: inherit;
  }
}

.option {
  white-space: normal;
  // Set focus on chrome
  outline-color: $select-focus;
}
