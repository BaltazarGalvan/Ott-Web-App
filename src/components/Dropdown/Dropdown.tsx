import React from 'react';
import classNames from 'classnames';

import styles from './Dropdown.module.scss';

type Props = {
  name: string;
  value: string;
  defaultLabel: string;
  options: string[];
  optionsStyle?: string;
  onChange: (event: React.ChangeEvent<HTMLSelectElement>) => void;
};

const Dropdown: React.FC<Props & React.AriaAttributes> = (
  {
    name,
    value,
    defaultLabel,
    options,
    onChange,
    optionsStyle,
    ...rest
  }: Props & React.AriaAttributes
) => {
  return (
    <div className={styles.dropdown}>
      <select className={styles.select} name={name} value={value} onChange={onChange} {...rest}>
        <option className={classNames(styles.option, optionsStyle)} value="">
          {defaultLabel}
        </option>
        {options.map((option) => (
          <option className={classNames(styles.option, optionsStyle)} key={option} value={option}>
            {option}
          </option>
        ))}
      </select>
      <span className="focus" />
    </div>
  );
};

export default Dropdown;
