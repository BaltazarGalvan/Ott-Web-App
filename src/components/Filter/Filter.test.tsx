import * as React from 'react';
import { render } from '@testing-library/react';

import '../../hooks/matchMedia.mock';
import Dropdown from './Filter';

const options = ['x', 'y', 'z'];

describe('<Dropdown>', () => {
  it('renders dropdown', () => {
    const { container } = render(
      <Dropdown
        name="categories"
        value="aa"
        defaultLabel="bb"
        options={options}
        setValue={(event) => event}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
