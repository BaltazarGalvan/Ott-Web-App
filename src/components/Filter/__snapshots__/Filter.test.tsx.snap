// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter> renders Filter 1`] = `
<div>
  <div
    class="filterModal"
  >
    <div
      class="header"
    >
      <div
        aria-label="close"
        class="iconButton"
        role="button"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="icon"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </g>
        </svg>
      </div>
      <h4>
        categories
      </h4>
    </div>
    <hr
      class="divider"
    />
    <div
      class="group"
    >
      <div
        class="menuButton"
        tabindex="0"
      >
        <span>
          x
        </span>
      </div>
      <div
        class="menuButton"
        tabindex="0"
      >
        <span>
          y
        </span>
      </div>
      <div
        class="menuButton"
        tabindex="0"
      >
        <span>
          z
        </span>
      </div>
      <div
        class="menuButton"
        tabindex="0"
      >
        <span>
          bb
        </span>
      </div>
    </div>
  </div>
  <div
    class="filterRow"
  >
    <button
      class="button"
    >
      <span>
        x
      </span>
    </button>
    <button
      class="button"
    >
      <span>
        y
      </span>
    </button>
    <button
      class="button"
    >
      <span>
        z
      </span>
    </button>
    <button
      class="button"
    >
      <span>
        bb
      </span>
    </button>
  </div>
</div>
`;
